### ISSUE TITLE
Autodiff backward panics on Conv1d(asymmetric padding) → GroupNorm → Conv1d for non-trivial input lengths

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#4799]
## Summary

`loss.backward()` panics when the forward graph is `Conv1d(asymmetric padding) → GroupNorm → Conv1d` and the input is long enough that the first conv's output has more than roughly 160 frames. The panic surfaces in both `NdArray` and `Flex` backends with different messages, so the root cause lives in burn's shared asymmetric-padding handling for `Conv1d`, not in any single backend.

Both conv layers use `PaddingConfig1d::Explicit(left, right)` with `right != left`. Forward produces correct output shapes on both backends; only backward fails.

## Minimal reproducer

```rust
use burn::nn::conv::Conv1dConfig;
use burn::nn::{GroupNormConfig, PaddingConfig1d};
use burn::prelude::*;
use burn::tensor::Distribution;
use burn::tensor::backend::AutodiffBackend;

fn run<B: AutodiffBackend>(device: &B::Device, input_length: usize) {
    let c1 = Conv1dConfig::new(1, 32, 10)
        .with_stride(10)
        .with_padding(PaddingConfig1d::Explicit(9, 0))
        .init::<B>(device);
    let gn = GroupNormConfig::new(8, 32).init::<B>(device);
    let c2 = Conv1dConfig::new(32, 64, 4)
        .with_stride(4)
        .with_padding(PaddingConfig1d::Explicit(3, 0))
        .init::<B>(device);

    let x = Tensor::<B, 3>::random([1, 1, input_length], Distribution::Default, device);
    let h = c1.forward(x);
    let h = gn.forward(h);
    let h = c2.forward(h);
    let _ = h.sum().backward();  // panics
}

#[test]
fn ndarray_panics() {
    use burn::backend::{Autodiff, NdArray};
    run::<Autodiff<NdArray>>(&Default::default(), 16_000);
}

#[test]
fn flex_panics() {
    use burn::backend::Autodiff;
    use burn_flex::Flex;
    run::<Autodiff<Flex>>(&Default::default(), 16_000);
}
```

## Expected behaviour

`loss.backward()` returns without panic and produces gradients with shapes matching the parameters (as it does on the equivalent `Conv1d + Conv1d` graph without the GroupNorm in between, and as it does at small input lengths).

### NdArray

```
panicked at ndarray-0.17.2/src/impl_methods.rs:591:22:
Slice end 1603 is past end of axis of length 1601
```

Stack trace points into `burn_autodiff::ops::backward::binary` feeding `burn_ndarray::ops::base::NdArrayOps::slice`, called from the autodiff step for `float_slice_assign`.

### Flex

```
panicked at burn-autodiff-0.21.0-pre.3/src/ops/base.rs:320:17:
Invalid broadcast shapes:
  Next grad shape Shape { dims: [1, 32, 1600] },
  Previous grad shape Shape { dims: [1, 32, 1598] }.
Expected the shape of the next grad to be 1.
```

## Input-length sweep (Flex, same two-conv + GroupNorm graph)

| `input_length` | `c1` output frames | backward |
| --: | --: | :-- |
| 160 | 16 | OK |
| 1 600 | 160 | OK |
| 16 000 | 1 600 | **panic** (1600 vs 1598) |
| 32 000 | 3 200 | **panic** (3200 vs 3198) |
| 64 000 | 6 400 | **panic** (16000 vs 15998) |
| 160 000 | 16 000 | **panic** (32000 vs 31998) |
| 320 000 | 32 000 | **panic** (33856 vs 33854) |
| 338 560 | 33 856 | **panic** (33856 vs 33854) |

The grad-shape mismatch is always exactly 2 frames, which matches `c2`'s `left_pad = 3` minus a likely off-by-one somewhere in the asymmetric-pad backward.

## Narrowing

- `c1 + c2` (no GroupNorm): **OK** on both backends.
- `c1 + GroupNorm` (no second conv): **OK** on both backends.
- `c1 + GroupNorm + c2` (above): **panics**.
- Same graph with symmetric padding (`PaddingConfig1d::Same` everywhere): **OK** (no asymmetric-pad path).

The critical ingredients are **asymmetric padding** on `c1` / `c2`, a non-identity op between them that materializes a new tensor in the autograd graph, and an input long enough to produce more than roughly 160 frames at the `c1` output. Removing any of the three makes backward succeed.

## Likely location

Wrapper in `burn_tensor::tensor::module::conv1d` (around burn-tensor 0.21.0-pre.3 line 47):

```rust
if let Some(padding_end) = padded_options.padding_end {
    let left = padded_options.options.padding[0];
    let right = padding_end[0];
    let padded = x.pad((left, right, 0, 0), PadMode::Constant(0.0));
    let zero_options = ConvOptions::new(..., [0], ...);
    // backend conv1d with padding=0 on the manually-padded tensor
}
```

The asymmetric-pad case routes through `x.pad(...)` + backend `conv1d(padding=0)`. The backward traversal through this pair appears to drop or miscompute the extra `left + right` padding slices, producing a gradient tensor whose `T` dim is `left + right` shorter than the upstream gradient expects.

## Versions

- `burn = "0.21.0-pre.3"`
- `burn-flex = "0.21.0-pre.3"`
- Rust 1.94
- macOS 15.4 (darwin 25.4)

## Workaround

None that preserves asymmetric padding. `PaddingConfig1d::Same` works but is symmetric.

**Comment by antimora** [comment]
Adding more backend coverage: **the same bug reproduces on the `tch` backend (LibTorch), on both `Mps` and `Cpu` devices**. Using the same reproducer graph (`Conv1d(asym pad) → GroupNorm → Conv1d(asym pad)`) and the same training-step pattern (forward + `loss.backward()`), raising from LibTorch's `gelu_backward` with an identical off-by-2 length mismatch on the time dimension:

Same reproducer pattern, driver code:
```rust
use burn::backend::Autodiff;
use burn::backend::libtorch::{LibTorch, LibTorchDevice};
use burn::nn::conv::Conv1dConfig;
use burn::nn::{GroupNormConfig, PaddingConfig1d};
use burn::prelude::*;
use burn::tensor::Distribution;

type B = Autodiff<LibTorch>;
// let device = LibTorchDevice::Mps;
let device = LibTorchDevice::Cpu;

// ... then: conv1 (PaddingConfig1d::Explicit(9, 0)) → GN → conv2 (PaddingConfig1d::Explicit(3, 0))
// → small MLP with GELU (or just conv1 + GN + gelu in a training step with input_length ≥ 16_000)
// → loss.backward() panics
```

### LibTorch MPS

```
thread 'main' panicked at tch-0.22.0/src/wrappers/tensor_generated.rs:9767:56:
called `Result::unwrap()` on an `Err` value: Torch("The size of tensor a (6354) must match the size of tensor b (6356) at non-singleton dimension 2
...
at::(anonymous namespace)::wrapper_MPS_gelu_backward(at::Tensor const&, at::Tensor const&, ...)
burn_tch::ops::activation::...::gelu_backward
burn_autodiff::ops::backward::unary
burn_autodiff::...::Gelu::backward
```

### LibTorch CPU

```
thread 'main' panicked at tch-0.22.0/src/wrappers/tensor_generated.rs:9767:56:
called `Result::unwrap()` on an `Err` value: Torch("The size of tensor a (6742) must match the size of tensor b (6744) at non-singleton dimension 2
...
at::(anonymous namespace)::wrapper_CPU_gelu_backward(at::Tensor const&, at::Tensor const&, ...)
burn_tch::ops::activation::...::gelu_backward
burn_autodiff::ops::backward::unary
burn_autodiff::...::Gelu::backward
```

Difference is always exactly 2, dimension 2 (time), independent of `input_length` (3388, 6354, 6742 all seen across backends). Whatever the upstream grad tensor's shape is, it's 2 frames shorter than what the downstream op expects — consistent with the original symptom path into `burn_ndarray::slice` and the `broadcast_shape` panic out of `burn_flex::ops::expand`.

### Summary matrix

| backend                 | forward | backward            |
| ----------------------- | :-----: | :-----------------: |
| NdArray (Autodiff)      | OK      | **panic** (slice)   |
| Flex (Autodiff)         | OK      | **panic** (broadcast) |
| LibTorch/MPS (Autodiff) | OK      | **panic** (gelu bw)  |
| LibTorch/CPU (Autodiff) | OK      | **panic** (gelu bw)  |
| WGPU (Autodiff)         | OK      | **OK**               |

Four backends panic with the same structural pattern (forward correct, backward shape mismatch of exactly `left_pad + right_pad - left_pad = right_pad` frames — wait actually always 2, which matches `k - 2` for the `k=4` second conv if the wrapper is subtracting an extra `k/2` somewhere). Only WGPU escapes, likely because its autodiff-forward path to the backend conv1d handles the manual `.pad(...)` differently.

That strongly suggests the root cause is in `burn_tensor::tensor::module::conv1d` (the asymmetric-padding wrapper that does `x.pad(...)` + backend `conv1d(padding=0)`), not in any individual backend's conv implementation. The grad-path accounting for the manual pad is the likely suspect.

Let me know if a full backend-parameterized reproducer (running the same `fn run<B: AutodiffBackend>(...)` over all four failing backends in a single test file) would be useful.
